From dcf7bae6c7b600364eb6ae9efb4e36895ef6875e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 10 Sep 2019 16:20:34 +0200
Subject: [PATCH 038/152] calendar-server: Temporarily ignore all deprecations
 in eds includes

Evolution draws in libsoup, which exposes deprecated types in its
API. While its headers have been fixed to guard the affected symbols,
those fixes aren't in our CI images yet.

Until we get a fixed version, just disable all deprecation warnings
during the include in order to not trip over "foreign" bugs.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/709
---
 src/calendar-server/calendar-sources.c            | 2 ++
 src/calendar-server/gnome-shell-calendar-server.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/calendar-server/calendar-sources.c b/src/calendar-server/calendar-sources.c
index aa3fa64f9..ed76c015c 100644
--- a/src/calendar-server/calendar-sources.c
+++ b/src/calendar-server/calendar-sources.c
@@ -29,7 +29,9 @@
 #include <string.h>
 #define HANDLE_LIBICAL_MEMORY
 #define EDS_DISABLE_DEPRECATED
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
 #include <libecal/libecal.h>
+G_GNUC_END_IGNORE_DEPRECATIONS
 
 #undef CALENDAR_ENABLE_DEBUG
 #include "calendar-debug.h"
diff --git a/src/calendar-server/gnome-shell-calendar-server.c b/src/calendar-server/gnome-shell-calendar-server.c
index 4eaaab4ac..0e074fbbc 100644
--- a/src/calendar-server/gnome-shell-calendar-server.c
+++ b/src/calendar-server/gnome-shell-calendar-server.c
@@ -35,7 +35,9 @@
 
 #define HANDLE_LIBICAL_MEMORY
 #define EDS_DISABLE_DEPRECATED
+G_GNUC_BEGIN_IGNORE_DEPRECATIONS
 #include <libecal/libecal.h>
+G_GNUC_END_IGNORE_DEPRECATIONS
 
 #include "calendar-sources.h"
 
-- 
2.23.0

